django/contrib/sessions/backends/base.py

```diff
@@ -301,7 +308,8 @@ def cycle_key(self):
         key = self.session_key
         self.create()
         self._session_cache = data
-        self.delete(key)
+        if key:
```

Comment:
I don't see any failing tests if these two lines are removed. Could you try to add one?